lib/Doctrine/ORM/Mapping/Driver/AnnotationDriver.php

```diff
@@ -70,6 +71,17 @@ public function __construct($reader, $paths = null)
         $this->reader = $reader;
 
         $this->addPaths((array) $paths);
+
+        if (! $reportFieldsWhereDeclared) {
+            Deprecation::trigger(
+                'doctrine/orm',
+                'https://github.com/doctrine/orm/pull/10455',
+                'In ORM 3.0, the AttributeDriver will report fields for the classes where they are declared. This may uncover invalid mapping configurations. To opt into the new mode also with the AnnotationsDriver today, set the "reportFieldsWhereDeclared" constructor parameter to true.',
```

Comment:
```suggestion
                'In ORM 3.0, the AttributeDriver will report fields for the classes where they are declared. This may uncover invalid mapping configurations. To opt into the new mode also with the AnnotationDriver today, set the "reportFieldsWhereDeclared" constructor parameter to true.',
```